File: Full_Stack_Capstone_Project/smartpizza_frontend/src/services/orderService.js
import apiClient from '../api/axiosClient';

export const orderService = {
  placeOrder: (payload) => apiClient.post('/orders', payload).then((res) => res.data),
  getMyOrders: () => apiClient.get('/orders/my-orders').then((res) => res.data),
  getOrderById: (id) => apiClient.get(`/orders/${id}`).then((res) => res.data),
  trackOrder: (id) => apiClient.get(`/orders/track/${id}`).then((res) => res.data),
  cancelOrder: (id) => apiClient.put(`/orders/${id}/cancel`).then((res) => res.data),
  reorder: (id) => apiClient.post(`/orders/${id}/reorder`).then((res) => res.data),
};

File: Full_Stack_Capstone_Project/smartpizza_backend/src/main/java/com/smartpizza/service/OrderService.java
package com.smartpizza.service;

import com.smartpizza.dto.request.PlaceOrderRequest;
import com.smartpizza.dto.response.CartResponse;
import com.smartpizza.dto.response.OrderResponse;
import com.smartpizza.dto.response.OrderTrackResponse;

import java.util.List;

public interface OrderService {

    OrderResponse placeOrder(String userEmail, PlaceOrderRequest request);

    List<OrderResponse> getMyOrders(String userEmail);

    OrderResponse getOrderById(String userEmail, Long orderId);

    OrderResponse cancelOrder(String userEmail, Long orderId);

    CartResponse reorder(String userEmail, Long orderId);

    OrderTrackResponse trackOrder(String userEmail, Long orderId);

    // ----- admin-only kitchen workflow -----

    List<OrderResponse> listAllOrders(String status);

    OrderResponse adminAccept(Long orderId);

    OrderResponse adminMarkPreparing(Long orderId);

    OrderResponse adminMarkBaked(Long orderId);
}
